sematic/plugins/metrics_storage/sql/sql_metrics_storage.py

```diff
@@ -100,8 +102,8 @@ def store_metrics(self, metric_points: Sequence[MetricPoint]) -> None:
     def get_metrics(self, labels: MetricsLabels) -> Tuple[str, ...]:
         with db().get_session() as session:
             metric_names = (
-                session.query(MetricLabel.metric_name)
-                .distinct(MetricLabel.metric_name)
+                session.query(MetricLabel.metric_name)  # type: ignore
+                .group_by(MetricLabel.metric_name)
```

Comment:
`distinct` in an `ON` clause is only supported for Postgres and is ignored for other DBs by SQLAlchemy. Luckily if the only column you are `select`ing is the same one you want distinct for, a `group_by` will still make sure you only consider distinct values (and is supported by other DB engines)